front/lib/api/assistant/messages.ts

```diff
@@ -272,14 +272,38 @@ async function getMaxRankMessages(
     };
   }
 
+  // First get the list of all visible user messages, so that we can filter agent messages answering them.
+  // Limit by the desired page size, that is the maximum number of user messages that can be displayed on a page.
+  const userMessages = await Message.findAll({
+    attributes: [
+      [Sequelize.fn("MAX", Sequelize.col("version")), "maxVersion"],
+      [Sequelize.fn("MAX", Sequelize.col("id")), "id"],
+    ],
+    where: {
+      ...where,
+      userMessageId: {
```

Comment:
I don't like the idea of making one extra round trip to the DB here... but I don't quite see any other way of doing it at the same time...

Reply:
I agree .. but I did not find another way to go.

Reply:
> This is not indexed. Should be added to index `["conversationId", "rank", "version"],`

Using `parentId: null` instead of `userMessageId: { [Op.not]: null }` , which is already indexed